felix/ip/ip_addr.go

```diff
@@ -101,6 +102,21 @@ func (a V6Addr) AsCIDR() CIDR {
 	}
 }
 
+// AsUint64Pair returns a pair of uint64 representing a V6Addr as there is
+// no native 128 bit uint type in go.
+func (a V6Addr) AsUint64Pair() (uint64, uint64) {
```

Comment:
Usually for representing v6 addresses, using `math/big.Int` is the way to go: https://pkg.go.dev/math/big

I think that will get you what you need?

Reply:
If so, might be worth getting both v4 and v6 logic to use it, so that it's easier to reuse code.

Reply:
Oh I briefly looked into it but dismissed it thinking that it wouldn't be compatible with the "bit scrubbing" needed (xor, shifting), but I think there's methods for that. Will do, thanks!